src/main/java/org/zalando/nakadi/service/timeline/TimelineService.java

```diff
@@ -197,6 +202,20 @@ private void switchTimelines(final Timeline activeTimeline, final Timeline nextT
         }
     }
 
+    private void scheduleTimelineCleanup(final Timeline timeline) throws InconsistentStateException {
+        try {
+            final EventType eventType = eventTypeCache.getEventType(timeline.getEventType());
+            final Long retentionTime = eventType.getOptions().getRetentionTime();
+            if (retentionTime == null) {
+                throw new InconsistentStateException("Event type should has information about its retention time");
+            }
+            final Date cleanupDate = new Date(System.currentTimeMillis() + retentionTime);
```

Comment:
There is a possibility to change retention time of the event type. CleanupUpAt has to be changed accordingly.